src/Features/CSharp/Portable/QuickInfo/CSharpSyntacticQuickInfoProvider.cs

```diff
@@ -22,17 +23,43 @@ namespace Microsoft.CodeAnalysis.CSharp.QuickInfo
     [ExtensionOrder(After = QuickInfoProviderNames.Semantic)]
     internal class CSharpSyntacticQuickInfoProvider : CommonQuickInfoProvider
     {
+        private readonly IDiagnosticAnalyzerService _diagnosticAnalyzerService;
+
         [ImportingConstructor]
         [SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification = "Used in test code: https://github.com/dotnet/roslyn/issues/42814")]
-        public CSharpSyntacticQuickInfoProvider()
+        public CSharpSyntacticQuickInfoProvider(IDiagnosticAnalyzerService diagnosticAnalyzerService)
         {
+            _diagnosticAnalyzerService = diagnosticAnalyzerService;
         }
 
         protected override async Task<QuickInfoItem?> BuildQuickInfoAsync(
             Document document,
             SyntaxToken token,
             CancellationToken cancellationToken)
         {
+            var pragmaWarningDiagnosticId = token.Parent switch
+            {
+                PragmaWarningDirectiveTriviaSyntax pragmaWarning => pragmaWarning.ErrorCodes.FirstOrDefault() as IdentifierNameSyntax,
+                IdentifierNameSyntax identifier when identifier.Parent is PragmaWarningDirectiveTriviaSyntax => identifier,
+                _ => null,
+            };
+            if (pragmaWarningDiagnosticId != null)
+            {
+                var diagnostics = await _diagnosticAnalyzerService.GetCachedDiagnosticsAsync(document.Project.Solution.Workspace, document.Project.Id, document.Id,
```

Comment:
this seems strange.   i would not expect you would need to go through this service.  but rather go through the compilation/semanticmodel for diagnostics for this node/span.

Reply:
I tried this in the first commit 91e7e192cc182c3f9f3c4cc382b520ef1fe79556 but this fall short for two reasons:

* the diagnostics did not contain any disabled warnings (I could solve this by removing the pragma from the syntactic tree and asking for a new semantic model), but
* even that semantic model did only contain compiler warnings. The StyleCop warnings were missing.